### PR TITLE
Re-enable xunit warning 2017

### PR DESCRIPTION
**Pull request by madmir**
Use specialized assertions for checking for elements in collections.

Contributes to #39697
cc: @stephentoub

### CODE REVIEW

File: src/System.Collections.Immutable/tests/ImmutableArrayBuilderTest.cs [hunk 1]
```diff
@@ -183,9 +183,9 @@ public void AddRangeDerivedBuilder()
         public void Contains()
         {
             var builder = new ImmutableArray<int>.Builder();
-            Assert.False(builder.Contains(1));
+            Assert.DoesNotContain(1, builder);
             builder.Add(1);
-            Assert.True(builder.Contains(1));
+            Assert.Contains(1, builder);
         }
```
Comment (stephentoub) [inline src/System.Collections.Immutable/tests/ImmutableArrayBuilderTest.cs]: This is a test for the Contains method; it should continue to use Contains directly.

File: src/System.Collections.Immutable/tests/ImmutableArrayTest.cs [hunk 2]
```diff
@@ -673,12 +673,12 @@ public void ContainsInt32(IEnumerable<int> source)
...
             {
                 int contained = Enumerable.Range(0, int.MaxValue).First(i => source.Contains(i));
-                Assert.True(array.Contains(contained));
+                Assert.Contains(contained, array);
                 Assert.True(((ICollection<int>)array).Contains(contained));
             }
 
             int notContained = Enumerable.Range(0, int.MaxValue).First(i => !source.Contains(i));
-            Assert.False(array.Contains(notContained));
+            Assert.DoesNotContain(notContained, array);
```
Comment (stephentoub) [inline src/System.Collections.Immutable/tests/ImmutableArrayTest.cs]: Ditto.

File: src/System.Collections.Immutable/tests/ImmutableHashSetBuilderTest.cs [hunk 3]
```diff
@@ -286,7 +286,7 @@ public void NullHandling()
             var builder = ImmutableHashSet<string>.Empty.ToBuilder();
             Assert.True(builder.Add(null));
             Assert.False(builder.Add(null));
-            Assert.True(builder.Contains(null));
+            Assert.Contains(null, builder);
```
Comment (stephentoub) [inline src/System.Collections.Immutable/tests/ImmutableHashSetBuilderTest.cs]: Ditto

**Review by stephentoub — CHANGES_REQUESTED** [review]
Thanks for working on this.

In starting to review the changes, I fear this change is actually moving things in the wrong direction.  The particular xunit rule means well, but for a codebase that actually implements a lot of Contains methods, we want tests for those Contains methods to continue using the explicit method the test was meant to test, and this change moves away from that.

My suggestion:
Revert all changes where the Contains method being used is actually part of the test, e.g. most of the changes to the tests for immutable collections.  Leave the rest of the changes, but then leave the rule disabled as well.

File: src/System.Collections.Immutable/tests/System.Collections.Immutable.Tests.csproj [hunk 4]
```diff
@@ -1,7 +1,7 @@
 <Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
     <ProjectGuid>{95DFC527-4DC1-495E-97D7-E94EE1F7140D}</ProjectGuid>
-    <NoWarn>0436</NoWarn>
+    <NoWarn>0436;NU5105;xUnit2017</NoWarn>
```
Comment (stephentoub) [inline src/System.Collections.Immutable/tests/System.Collections.Immutable.Tests.csproj]: This can be reverted.

**Review by stephentoub — APPROVED** [review]
Thanks!

File: src/System.Collections.Immutable/tests/System.Collections.Immutable.Tests.csproj (hunk 4 above)
Comment (madmir) [reply src/System.Collections.Immutable/tests/System.Collections.Immutable.Tests.csproj]: Sure, I was considering it as one of the alternatives for disabling the rule and then it went in by mistake.

**Pull request merged**